.github/workflows/check-broken-links.yml

```diff
@@ -0,0 +1,39 @@
+name: Check Dead Links
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  check-deadlinks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+
+      - name: Install markdown-link-check
+        run: npm install -g markdown-link-check
```

Comment:
TBH I would use a more specific tool as [htmltest](https://github.com/wjdp/htmltest) ([here's a guide on how to integrate on github action](https://robb.sh/posts/check-links-in-hugo-with-htmltest/)), because I'm not sure that checking in the markdowns will be enough, it would be better to check the built site (because Hugo does some additional interpolation on the internal links)